### PR TITLE
Cleanup : validatePromise ranamed to isNotValidPromise with inverted state + tests

### PR DESCRIPTION
**Pull request by doom369**
Motivation:

Result of `validatePromise()` is always inverted with `if (!validatePromise())`.

Modification:

`validatePromise()` renamed to `isNotValidPromise()` and now returns inverted state so you don't need to invert state in conditions. Also name is now more meaningful according to returned result.
Added more tests for `validatePromise` corner cases with Exceptions.

Result:

Code easier to read. No need in inverted result.

### CODE REVIEW

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 1]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+        group.register(pipeline2.channel());
+
+        ChannelPromise promise2 = pipeline2.channel().newPromise();
+        pipeline.close(promise2);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testUnexpectedVoidChannelPromise() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: `.sync()`

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 2]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+
+        ChannelPromise promise = new VoidChannelPromise(pipeline.channel(), false);
+        pipeline.close(promise);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testUnexpectedVoidChannelPromiseCloseFuture() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: `.sync()`

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 3]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
         assertTrue(future.isCancelled());
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void testWrongPromiseChannel() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: `.sync()`

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 4]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+
+        ChannelPipeline pipeline2 = new LocalChannel().pipeline();
+        group.register(pipeline2.channel());
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: `.sync()`

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 5]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+
+        ChannelPromise promise2 = pipeline2.channel().newPromise();
+        pipeline.close(promise2);
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: close the two channels in a finally block

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 6]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testUnexpectedVoidChannelPromise() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
+
+        ChannelPromise promise = new VoidChannelPromise(pipeline.channel(), false);
+        pipeline.close(promise);
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: close the channel in a finally block

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java [hunk 7]
```diff
@@ -504,6 +504,36 @@ public void testCancelClose() throws Exception {
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testUnexpectedVoidChannelPromiseCloseFuture() throws Exception {
+        ChannelPipeline pipeline = new LocalChannel().pipeline();
+        group.register(pipeline.channel());
+
+        ChannelPromise promise = (ChannelPromise) pipeline.channel().closeFuture();
+        pipeline.close(promise);
```
Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: close the channel in a finally block

File: transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java (hunk 3 above)
Comment (doom369) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: @normanmaurer hm, I copied this from your test and it doesn't have sync in it :).

Comment (normanmaurer) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: I think we should sync on all of them... not super important tho 👍

Comment (doom369) [inline transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java]: ok

**Comment by doom369** [comment]
@normanmaurer done

**Comment by normanmaurer** [comment]
@doom369 thanks a lot!

Cherry-picked into 4.1 (64838f150584385f6d8608ae11959836c5a8eaf9) and 4.0 (59aa57cc0319547981d650521599dd7b97853df9)
